HERMES-SOC/hermes_core
# Remove TimeData Dependency on CDF C Library

This PR removes the requirement to have the CDF C libraries installed in order to create and manipulate `TimeData` containers. This will support teams down the line to use data formats other than CDF for saving and loading data into the `TimeData` container. 

Closes #59

*To Test this PR* 

1. Do what you would normally do to checkout the PR branch locally
2. Disable the CDF Libraries in your Dev Container's Bash session
    ```sh
    sudo mv /usr/local/cdf /usr/local/tmp_cdf
    ```
3. In a Python interface of your choice, verify that you can create a `TimeData` container and that attributes can be derived without using the `spacepy.pycdf` library.
    ```python
    import numpy as np
    from astropy.timeseries import TimeSeries
    import astropy.units as u
    from hermes_core.timedata import TimeData

    # Create Global Attributes
    global_attrs_template = TimeData.global_attribute_template("eea", "l1", "0.0.1")

    # Create a Test TimeSeries
    ts = TimeSeries(
        time_start="2016-03-22T12:30:31",
        time_delta=3 * u.s,
        data={"Bx": u.Quantity([1, 2, 3, 4], "gauss", dtype=np.uint16)},
    )

    # Create a Test Data Container
    cont = TimeData(ts, meta=global_attrs_template)

    # You can now verify that the container can be manipulated and 
    # attributes have been derived without the CDF Libraries

    # If you try to run 
    # `cont.save()`
    # You will see that the data container still cannot be saved without the CDF libraries
    ```
4. Re-Enable the CDF Libraries in your Dev Container
    ```sh
    sudo mv /usr/local/tmp_cdf /usr/local/cdf 
    ```
5. Hit Approve :)
